src/wasm/kotlin/java/arcs/Addressable.kt

```diff
@@ -0,0 +1,30 @@
+package arcs
+
+/**
+ * Any object implementing this interface can be converted into a stable identifier. This
+ * could a platform specific pointer, identity hashcode, or id as long as it is guaranteed unique.
+ */
+interface Addressable
+
+typealias Address = Int
+
+internal val address2Addressable = kotlin.collections.mutableMapOf<Address, Addressable>()
+internal val addressable2Address = kotlin.collections.mutableMapOf<Addressable, Address>()
+
+internal var nextAddress = 1;
+
+fun Addressable.toAddress(): Address {
+    if (addressable2Address[this] != null) {
+        return addressable2Address[this]!!
```

Comment:
FMI: what does `!!` do? I'm guessing it's a way of forcing this to be non-null, but I'm curious why it's needed if we have the if-statement above.

Reply:
Jason? Why does the guard not work? Cameron, you're right in what it's doing, but it's not supposed to be needed, only my IDE was complaining. 

That is, normally I thought kotlin would allow if (a != null) { return a } to be returned from a non-nullable function without coercion. 

Jason: What's the idiomatic for this? addressable2Address[this]?.let { it } ?: { code to construct the mapping }?

Reply:
Your suggestion is idiomatic as far as I understand. 
This could also be an appropriate place to have an if-else expression: 

```
return if (addressable2Address[this] != null) addressable2Address[this] 
           else { /* ... */ }
```